### PR TITLE
#52 - Mock landing page

### LINKED ISSUE #52: Mock Landing Page
**User Story**
As a user, I want to be able to prototype a landing page for the application with _mocked_ sign-up/sign-in functionality

**Acceptance Criteria**
* Design as per figma prototype
* Stub sign-up button (does nothing)
* Mock sign-in page arbitrarily allowing all users

**Additional Context**
This is purely for **demonstration purposes**; the real signup/login functionality will have to be implemented at a later date.

**Dependencies**
N/A

### PR DESCRIPTION
**Pull request by sannan06**
**Issue:**
Closes #52 

**Solution:**
Added a LandingPage component designed as per Figma prototype, with mocked "Sign up' button.
Added react router so default route is /signup (i.e. landing page), and added a route for '/home/

**Test Method:** 
Manually checked styling of elements.
Manually checked routing logic
Added unit test that clicking "Sign in" routes to "/login"

**Risk:**
Replaces default view with the landing page, and places the actual application behind "/home". Note, it is possible to automatically route to "/home" if a user is logged in already (conditionally redirecting)

**Reviewed By:**
[Edit the message as to who reviewed it here]

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+                    <LandingPage />
+                </Route>
+
+                <Route exact path="/home">
+                    {/* Replace following component with actual Home component when created */}
+                    <Header />
+                </Route>
+            </Switch>
+        </BrowserRouter>
     );
```
Comment (kcho9906) [inline client/src/App.js:26]: Can do check for whether or not the user is signed in or not later down the track but this is good for the mock

File: client/src/components/LandingPage/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    },
+});
+
+export default function LandingPage() {
+    const classes = useStyles();
+    const history = useHistory();
+
+    return (
+        <div className={styles.page}>
+            <div className={styles.container}>
```
Comment (kcho9906) [inline client/src/components/LandingPage/index.js:40]: any reason we have two divs?

**Review by kcho9906 — APPROVED** [review]

File: client/src/components/LandingPage/index.js (hunk 2 above)
Comment (sannan06) [reply client/src/components/LandingPage/index.js:40]: Outer `.page` is full height/width of window & sets blue background colour; inner `.container` is a 80vw container for the landing page elements.

**Review by ggao007 — APPROVED** [review]

**Review by tommyshi55 — APPROVED** [review]
Looks good to me! Just need to rebase onto main and fix the conflict before it can be merged :)

File: client/src/App.test.js [hunk 3]
```diff
@@ -1,10 +0,0 @@
-import React from "react";
-import { render, screen } from "@testing-library/react";
-import { test, expect } from "@testing-library/jest-dom";
-import App from ".";
-
-test("renders learn react link", () => {
-    render(<App />);
-    const linkElement = screen.getByText(/learn react/i);
-    expect(linkElement).toBeInTheDocument();
-});
```
Comment (Denijar) [inline client/src/App.test.js:10]: This file should not be deleted. Instead, update the snapshot (by running `npm test` then pressing u to update) and altering the tests to match the new functionality.

File: client/src/components/LandingPage/index.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                <div className={styles.focusIntro}>
+                    <img className={styles.focusLogo} src={FocusLogo} alt="FOCUS" />
+                    <p>
+                        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc blandit in
```
Comment (Denijar) [inline client/src/components/LandingPage/index.js:44]: Can we have some real text here: for example here is the description text from the Wiki: **Daily Focus is a dashboard web app that surfaces vital at-a-glance information to enable you to know what you need to know and get on with your day. The dashboard integrates with a number of different data providers and services, and the display of this data can be configured in the form of widgets to best suit your needs.**

File: client/src/components/LandingPage/index.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                        id="outlined-basic"
+                        variant="outlined"
+                        label="Password"
+                    />
+
+                    <Button
+                        variant="contained"
+                        color="primary"
+                        classes={{ root: classes.buttonRoot, label: classes.label }}
+                        onClick={() => history.push("/home")}
```
Comment (Denijar) [inline client/src/components/LandingPage/index.js:74]: Can we add an issue to the repo which is for linking this to the signup endpoint.

**Review by Denijar — CHANGES_REQUESTED** [review]

**Comment by Denijar** [comment]
Landing page looks extremely professional. Well done on the styling.

File: client/src/components/LandingPage/index.js (hunk 4 above)
Comment (Denijar) [reply client/src/components/LandingPage/index.js:44]: (or just the first sentence if that description is too long)

**Comment by sannan06** [comment]
@Denijar I can't merge until your requested changes are approved

**Review by Denijar — APPROVED** [review]
Looks great, nice job.

**Pull request merged**